### PR TITLE
Make `hasdoc` public

### PR DESCRIPTION
**Pull request by jariji**
Making `Docs.hasdoc` `public` per @LilithHafner  https://github.com/JuliaLang/julia/pull/52139#issuecomment-1874748877.

### CODE REVIEW

File: base/docs/Docs.jl [hunk 1]
```diff
@@ -66,6 +66,7 @@ using .Base: RefValue, mapany
 import ..CoreDocs: lazy_iterpolate
 
 export doc
+public hasdoc
```
Comment (stevengj) [inline base/docs/Docs.jl:69]:
```suggestion
public hasdoc, undocumented_names
```

Comment (jariji) [reply base/docs/Docs.jl:69]: I was holding off until hearing from Triage about whether `undocumented_names` belongs in `Docs` vs `Test`/`Aqua` as in the comment in that thread. Please comment there if you have a view on that.

**Comment by stevengj** [comment]
Should it simply be exported?

(Note that even if you add `export hasdoc` here, it still won't be re-exported by `Base` by default.)

**Comment by stevengj** [comment]
I also fixed the docstring to refer to "public" symbols rather than "exported" symbols, following https://github.com/JuliaLang/julia/pull/52413#issuecomment-1873032909 by @LilithHafner

**Comment by LilithHafner** [comment]
I don't see much advantage of exporting from a submodule (historically, I'm guessing the primary uses were to mark the symbols public and when a symbol was previously defined in Base and moved to a submodule)

**Comment by jariji** [comment]
In favor of `export`:

- people who do `using Base.Docs` can get the convenience they want.
- `export` is the most common way of making something public, so without a particular reason it seems good to keep with the normal practice. It's nice to have a single way of doing things that works as often as possible.

**Comment by stevengj** [comment]
If we are going to export `doc` and `@var` and `@text_str` (some of which aren't even documented), it doesn't make sense to me not to export `hasdoc` and `documented_symbols` (which are much more likely to be used outside of Base).   And, after all, people can do `using Base.Docs` or `using `.Docs` as @jariji pointed out.

**Comment by stevengj** [comment]
(After this, I think we need to make a concerted effort to get `@test isempty(undocumented_symbols(Foo))` to pass for stdlib modules and documented submodules.)

**Comment by stevengj** [comment]
Bump.

**Comment by stevengj** [comment]
@LilithHafner, `undocumented_names` was already changed (at your request) to include all public symbols, including macros and operators, that don't begin with `#` (to exclude generated internal symbols, ala `gensym`), don't you remember?  #52924 52743

**Comment by LilithHafner** [comment]
> don't you remember?

Clearly not, lol. You are quite right.

**Pull request merged**